tests/rptest/tests/timequery_test.py

```diff
@@ -458,6 +462,57 @@ def query_slices(tid):
 
         assert not any([e > 0 for e in errors])
 
+    @cluster(num_nodes=4)
+    def test_timequery_with_spillover_gc_delayed(self):
+        self.set_up_cluster(cloud_storage=True,
+                            batch_cache=False,
+                            spillover=True)
+        total_segments = 16
+        record_size = 1024
+        base_ts = 1664453149000
+        msg_count = (self.log_segment_size * total_segments) // record_size
+        local_retention = self.log_segment_size * 4
+        topic_retention = self.log_segment_size * 8
+        topic, timestamps = self._create_and_produce(self.redpanda, True,
+                                                     local_retention, base_ts,
+                                                     record_size, msg_count)
+
+        # Confirm messages written
+        rpk = RpkTool(self.redpanda)
+        p = next(rpk.describe_topic(topic.name))
+        assert p.high_watermark == msg_count
+
+        # If using cloud storage, we must wait for some segments
+        # to fall out of local storage, to ensure we are really
+        # hitting the cloud storage read path when querying.
+        wait_for_local_storage_truncate(redpanda=self.redpanda,
+                                        topic=topic.name,
+                                        target_bytes=local_retention)
+
+        # Update the retention to trigger archive retention mechanisms but
+        # keep firewall blocked to prevent garbage collection.
+        with firewall_blocked(
+                self.redpanda.nodes,
+                self.si_settings.cloud_storage_api_endpoint_port):
+            self.client().alter_topic_config(topic.name, 'retention.bytes',
+                                             topic_retention)
+            self.logger.info("Waiting for start offset to advance...")
+            wait_until(
+                lambda: next(rpk.describe_topic(topic.name)).start_offset > 0,
+                timeout_sec=120,
+                backoff_sec=5,
+                err_msg="Start offset did not advance")
+
+            # Query below valid timestamps the offset of the first message.
+            kcat = KafkaCat(self.redpanda, num_retries=2)
+
+            # This fails for now until garbage collection is run.
+            # Tech debt.
+            with expect_exception(subprocess.CalledProcessError,
+                                  lambda e: True):
+                offset = kcat.query_offset(topic.name, 0, timestamps[0] - 1000)
+                assert offset > timestamps[0]
```

Comment:
is there a ticket to track this, linking this snippet and the timequery in remote_partition.cc?